aFWall/src/main/java/dev/ukanth/ufirewall/preferences/SecPreferenceFragment.java

```diff
@@ -258,6 +280,21 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,
             //passOption = "p" + index;
 
             //currentPosition = index;
+
+            // check if device support fingerprint, if so check if one fingerprint already existed at least
+            if(FingerprintUtil.isAndroidSupport()){
+
+                checkFingerprintDeviceSupport();
+
+            }else{
+
+                if(itemList != null) {
+
+                    itemList.setValueIndex(0);
```

Comment:
Why are we resetting to no password if FingerPrint is not supported. What about the usecase if user was using someother option ?